### DISCUSSION TITLE
How to query collections from custom plugin?

### DISCUSSION THREAD

**Discussion by podeig** [strapi/strapi#6970] [Legacy Topics]
I need to build my own plugin.
I want to fetch data from my collection 'subscription' but I cannot find how to do it.

I tried like 

    const subscriptions = strapi.query('subscription').find({});

but strapi.query is undefined.

```
import React, { memo } from 'react';
import pluginId from '../../pluginId';

const HomePage = () => {
  const renderEmails = () => {
      // Here I want to fetch emails from my collection and display it
      return null;
  }

  return (
    <div>
      <h1>{pluginId}&apos;s HomePage</h1>
      <p>Happy coding</p>
      {renderEmails()}
    </div>
  );
};

export default memo(HomePage);
```

**Comment by derrickmehaffy** [comment]
Hello,

I see you are wanting to ask a question, we are currently planning to decomission our GitHub Discussions in early 2021 and we don't want to lose the great questions asked by our community.

Questions should be directed to [our forum](https://forum.strapi.io/). Please see the following contributing guidelines for asking a question [here](https://github.com/strapi/strapi/blob/master/CONTRIBUTING.md#reporting-an-issue).
